neat/trees/greenstree.py

```diff
@@ -133,12 +120,21 @@ def _setImpedanceDistal(self):
         Set the boundary condition at the distal end of the segment
         """
         if len(self.child_nodes) == 0:
-            self.z_distal = np.infty*np.ones(len(self.z_m)) if self.g_shunt < 1e-10 else \
-                            1. / self.g_shunt
+            z_aux = np.ones(self.z_m.shape, dtype=float)
+
+            if self.g_shunt > 1e-10:
+                z_aux /= self.g_shunt
+            else:
+                z_aux *= np.infty
```

Comment:
```suggestion
                z_aux = np.infty
```
i guess *infity is the same as setting it to infinity but is (for me) clearer

Reply:
This syntax doesn't work because then z_aux is not an array anymore

Reply:
hmm, what about `np.full_like(z_aux, np.infty)`? for me this is clearer, but it's a subjective matter I think.